language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSAnnotationCache.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver;
+
+import org.ballerinalang.model.elements.PackageID;
+import org.wso2.ballerinalang.compiler.tree.BLangAnnotation;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Annotation cache for Language server.
+ * 
+ * Note: Annotation cache should be synced with the LS Package Cache
+ * 
+ * @since 0.970.0
+ */
+class LSAnnotationCache {
```

Comment:
So purpose of this cache is to isolate annotations from packages or are they co dependent ?

Reply:
Annotations are being isolated with the annotation cache since the package cache is more coarse-grained and the annotation cache is more fine-grained